packages/models-library/src/models_library/api_schemas_resource_usage_tracker/credit_transactions.py

```diff
@@ -11,13 +12,11 @@
 
 class WalletTotalCredits(BaseModel):
     wallet_id: WalletID
-    available_osparc_credits: Decimal
-
-    @field_validator("available_osparc_credits")
-    @classmethod
-    def ensure_rounded(cls, v):
-        return round(v, 2)
-
+    available_osparc_credits: Annotated[
+        Decimal, 
+        BeforeValidator(lambda x: round(x, 2)),
+        PlainSerializer(float, return_type=float, when_used='json')
```

Comment:
The `PlainSerializer` is used here to serialize as `float` the `Decimal` field that from Pydantic v2 would be returned as `str` in the `JSON` serialization. Without it the OpenAPI schema would have a field of type `string` instead of `number`, changing the interface.

Reply:
If unit tests are passing, then it should be fine (we have some tests for this Decimal checks)

Reply:
@matusdrobuliak66  I’m quite skeptical about the changes in these `Decimal`   ... should be test it further  ?:-(